enhancements/20190903-SRIOV-GA.md

```diff
@@ -0,0 +1,178 @@
+title: SR-IOV GA
+authors:
+  - "@zshi-redhat"
+  - "@pliurh"
+  - "@s1061123"
+reviewers:
+  - "@fepan"
+  - "@dougbtv"
+  - "@vpickard"
+  - "@dcbw"
+  - "@squeed"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2019-09-03
+last-updated: 2019-09-03
+status: implementable
+---
+
+# Graduate SR-IOV to GA
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+A proposal to graduate SR-IOV support from Tech Preview to GA.
+
+SR-IOV networks are enabled by Multus on secondary-onward pod interfaces.
+It is supported as Tech Preview on OpenShift 4.1 and 4.2 releases.
+This proposal is to make SR-IOV Generally Available in 4.3.
+
+## Motivation
+
+* SR-IOV is currently Technical Preview, but customers need to build supported
+  solutions on the technology.
+
+### Goals
+
+- Enhance Multus to support runtimeConfig for IP assignment
+- Enhance Multus to handle default route overwrite
+- Add admission controller for SR-IOV Network Operator
+- Compare SR-IOV network performance w/wo NUMA affinity
+- Support SR-IOV QoS flags
+- Add SR-IOV Metrics and Alerting
+- Support both RHEL and RHCOS worker nodes
+
+### Non-Goals
+
+- GA userspace (dpdk) workload support
+
+## Proposal
+
+### Implementation Details
+
+#### Multus
+
+Static IPAM plugin is used to assign IPv4 and IPv6 addresses statically to
+container interface. It supports receiveing IPv4/v6 address configuration from
+both CNI config and runtimeConfig. However, Multus only supports passing IP
+configuration via CNI config. This proposal is to add runtimeConfig support
+in Multus CNI.
+
+##### Default Route Overwrite
+
+Every pod comes with a default route added via openshift-sdn. It is not
```

Comment:
This just sounds like https://github.com/nokia/danm

Reply:
@squeed Is it because that with `default route overwrite`, Multus is doing too much other than just calling delegated plugins which sounds like a thick plugin such as danm?

Reply:
@squeed btw, we're going to use https://github.com/redhat-nfvpe/cni-route-override to override CNI route (except default route). For default route, we are proposing 'gateway' feature in NPWG.

If you think that multus includes route manipulation inside it and getting thick plugin, that is not correct, actually. We're still designing as minimum as possible.

Reply:
Okay, that should work as well. Could you provide some example configuration? It's important that we design something that will work with CNI CHECK.

Reply:
Here is the example config:
https://github.com/redhat-nfvpe/cni-route-override#example-configuration

Reply:
will add a default route overwrite example later, sth like this:
```
kind: Pod
metadata:
  name: my-pod
  namespace: my-namespace
  annotations:
    k8s.v1.cni.cncf.io/networks: |
      [
        { "name":"net-a" },
        {
          "name":"net-b",
          "default-route": ["10.0.0.1"]
        }
      ]
```

Reply:
also paste the link for default route overwirte proposal: https://docs.google.com/document/d/1pqbWYtFdEYyXd1cXuyvq_J5SRnhiXCRptXho63gI4A8/edit